docs/yul.rst

```diff
@@ -1124,6 +1124,10 @@ regular strings in native encoding. For code,
 
 Above, ``Block`` refers to ``Block`` in the Yul code grammar explained in the previous chapter.
 
+.. note::
+
+    An object with a name that ends in ``_deployed`` is treated as runtime code by the Yul optimizer.
```

Comment:
We should say that the only consequence of this is the tuning of the gas cost heuristic of the optimizer.